src/RwaSpell.sol

```diff
@@ -0,0 +1,204 @@
+    function tell(bytes32) external;
+    function cure(bytes32) external;
+    function cull(bytes32) external;
+    function good(bytes32) external view;
+}
+
+interface RwaConduitLike {
+    function wards(address) external returns (uint256);
+    function can(address) external returns (uint256);
+    function rely(address) external;
+    function deny(address) external;
+    function hope(address) external;
+    function nope(address) external;
+    function bud(address) external returns (uint256);
+    function kiss(address) external;
+    function diss(address) external;
+    function pick(address) external;
+    function push() external;
+}
+
+contract SpellAction {
+    // KOVAN ADDRESSES
+   //
+    // The contracts in this list should correspond to MCD core contracts, verify
+    // against the current release list at:
+    //     https://changelog.makerdao.com/releases/mainnet/1.1.4/contracts.json
+
+    ChainlogAbstract constant CHANGELOG = ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+
+    address constant MCD_VAT         = 0xbA987bDB501d131f766fEe8180Da5d81b34b69d9;
+    address constant MCD_CAT         = 0xdDb5F7A3A5558b9a6a1f3382BD75E2268d1c6958;
+    address constant MCD_JUG         = 0xcbB7718c9F39d05aEEDE1c472ca8Bf804b2f1EaD;
+    address constant MCD_SPOT        = 0x3a042de6413eDB15F2784f2f97cC68C7E9750b2D;
+    address constant MCD_END         = 0x24728AcF2E2C403F5d2db4Df6834B8998e56aA5F;
+    address constant ILK_REGISTRY    = 0xedE45A0522CA19e979e217064629778d6Cc2d9Ea;
+
+    address constant RWA001_GEM      = 0xE255c837f9363213dFD716C8Ed874A8B33e6a553;
+    address constant MCD_JOIN_RWA001 = 0x70efd9582843043d5055e558532F700e29B4Ff1a;
+    address constant MCD_FLIP_RWA001 = 0x8022Fd8a28A3acCE3C45bBbca8d3B7B972700153;
+    address constant PIP_RWA001      = 0x51486fbD0e669b48eA28Dee273Fac5F89402f982;
+    address constant RWA_URN         = 0xC6172B516f265dF53123F052FAAEB2AD63e49df7;
+    address constant RWA_CONDUIT     = 0x0CF836924fD65Af0DE42294c8e9FAcCC19A384Dc;
+
+    // this is set to 10million right now
+    address constant PIP             = 0x0318D82C3b2a23d993dcE881aada122f311ca901;
+
+    // may be able to assign these as immutable
+    // try constants
+    DSValueAbstract constant pippip = DSValueAbstract(PIP);
+
+    RwaLiquidationLike constant oracle        = RwaLiquidationLike(PIP_RWA001);
+
+    uint256 constant SIX_PCT_RATE    = 1000000001847694957439350562;
+
+    // precision
+    uint256 constant public THOUSAND = 10 ** 3;
+    uint256 constant public MILLION  = 10 ** 6;
+    uint256 constant public WAD      = 10 ** 18;
+    uint256 constant public RAY      = 10 ** 27;
+    uint256 constant public RAD      = 10 ** 45;
+    
+    function execute() external {
+
+        ////////////////////////////////////////////////////////////////////////////////
+        // RWA-001 collateral deploy
+
+        // Set ilk bytes32 variable
+        bytes32 ilk = "RWA-001";
```

Comment:
Move this out to a constant `ILK_RWA001` to allow for multiple ilk deploys in the same method.